src/Xamarin.Android.Build.Tasks/Tasks/ProcessAssemblies.cs

```diff
@@ -47,53 +47,60 @@ public override bool RunTask ()
 				}
 			}
 
-			foreach (var assembly in InputAssemblies) {
-				if (!File.Exists (assembly.ItemSpec)) {
-					Log.LogDebugMessage ($"Skipping non-existent dependency '{assembly.ItemSpec}'.");
-					continue;
-				}
-				using (var pe = new PEReader (File.OpenRead (assembly.ItemSpec))) {
-					var reader = pe.GetMetadataReader ();
-					var module = reader.GetModuleDefinition ();
-					var mvid = reader.GetGuid (module.Mvid);
-					if (!output.ContainsKey (mvid)) {
-						output.Add (mvid, assembly);
-
-						// Set metadata, such as %(FrameworkAssembly) and %(HasMonoAndroidReference)
-						string packageId = assembly.GetMetadata ("NuGetPackageId");
-						bool frameworkAssembly = packageId.StartsWith ("Microsoft.NETCore.App.Runtime.") ||
-							packageId.StartsWith ("Microsoft.Android.Runtime.");
+			// Group by assembly file name
+			foreach (var group in InputAssemblies.Where (Filter).GroupBy (a => Path.GetFileName (a.ItemSpec))) {
+				// Get the unique list of MVIDs
+				var mvids = new HashSet<Guid> ();
+				bool? frameworkAssembly = null, hasMonoAndroidReference = null;
+				foreach (var assembly in group) {
+					using (var pe = new PEReader (File.OpenRead (assembly.ItemSpec))) {
+						var reader = pe.GetMetadataReader ();
+						var module = reader.GetModuleDefinition ();
+						var mvid = reader.GetGuid (module.Mvid);
+						mvids.Add (mvid);
+
+						// Calculate %(FrameworkAssembly) and %(HasMonoAndroidReference) for the first
+						if (frameworkAssembly == null) {
+							string packageId = assembly.GetMetadata ("NuGetPackageId");
+							frameworkAssembly = packageId.StartsWith ("Microsoft.NETCore.App.Runtime.") ||
```

Comment:
Can `packageId` be `null`?  Should we use `?.StartsWith(…)`?

Reply:
`GetMetadata` should return `string.Empty` if not found:

* https://github.com/dotnet/msbuild/blob/7452552ce911efdca6aea5a189c409f083db7bc7/src/Utilities/TaskItem.cs#L116-L140
* https://github.com/dotnet/msbuild/blob/7452552ce911efdca6aea5a189c409f083db7bc7/src/Utilities/TaskItem.cs#L407-L428

But we could add a `null` check, as it is still possible.